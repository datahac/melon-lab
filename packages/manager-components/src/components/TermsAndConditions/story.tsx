import { storiesOf } from '@storybook/react';
import React from 'react';
import TermsAndConditions from './index';

const data = {
  networkId: '1',
};

storiesOf('Components|Terms And Conditions', module)
  .add('Default', () => {
    return <TermsAndConditions />;
  })
  .add('Network 1', () => {
    return <TermsAndConditions {...data} />;
  });
