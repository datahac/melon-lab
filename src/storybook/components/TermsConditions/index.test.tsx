import React from 'react';
import TermsAndConditions from './index';

const data = {
  networkId: '1',
  sign: () => null,
};

describe('TermsAndConditions', () => {
  const defaultElement = <TermsAndConditions {...data} />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly without networkId', () => {
    wrapper.setProps({ networkId: undefined });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
