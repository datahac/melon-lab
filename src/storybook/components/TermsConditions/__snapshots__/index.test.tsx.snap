// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsAndConditions should render correctly 1`] = `
<div
  className="conditions"
>
  <h4>
    Disclaimer for the use of the Melon Protocol (“Disclaimer”)
  </h4>
  <p>
    Please read the complete version of the terms and conditions
     
    <a
      href="https://github.com/melonproject/contribution/blob/master/Melonport_Second%20Contribution%20Period_FINAL.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
    </a>
  </p>
  <p>
    <strong>
      THIS SOFTWARE IS INTENDED FOR TESTING PURPOSES ONLY.
    </strong>
  </p>
  <p>
    <strong>
      User confirms to have read the full terms of the second contribution period & first capsule (Paros) carefully before continuing to the next step. Agreement to this Disclaimer is subject to being represented by a secure elliptic curve digital signature using the IPFS hash of the Disclaimer.
    </strong>
  </p>
  <p>
    <strong>
      DEPLOYMENT OF MELON PROTOCOL IN A MAINNET ENVIRONMENT ONLY FOR TESTING PURPOSES
    </strong>
    : Melonport AG (the “Company”) has developed a Smart-Contract-System (the “Melon Protocol”) which enables users, to manage crypto assets. THE SOFTWARE DEPLOYMENT OF THE CURRENT VERSION OF THE MELON PROTOCOL TO THE ETHEREUM MAINNET IS MADE FOR TESTING PURPOSES IN A LIVE ENVIRONMENT ONLY and within the clear framework set out by the
     
    <a
      href="https://github.com/melonproject/contribution/blob/master/Melonport_Second%20Contribution%20Period_FINAL.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      first capsule (Paros) and second contribution period terms
       
    </a>
    .
  </p>
  <p>
    <strong>
      NO WARRANTY
    </strong>
    : THE USER THEREFORE UNDERSTANDS AND ACKNOWLEDGES THAT AS THE MELON PROTOCOL IS OF EXPERIMENTAL NATURE AND DEPLOYED FOR TESTING MODE ONLY, ANY ETH OR OTHER CRYPTO ASSETS USED TO INTERACT WITH THE MELON PROTOCOL ARE AT RISK OF BEING LOST INDEFINITELY. User understands and accepts that the Melon Protocol is provided “as is” and any expressed or implied warranties of merchantability and fitness for a particular purpose are disclaimed. In particular, User acknowledges that the Melon Protocol, even though developed in accordance to the state of the art, is likely to contain bugs, defects, or errors (including any bug, defect, or error relating to or resulting from the display, manipulation, processing, storage, transmission, or use of data) that materially and adversely affects the use, functionality, or performance of the Melon Protocol, or any product or system containing or used in conjunction with the Melon Protocol.
  </p>
  <p>
    <strong>
      NO LIABILITY OF THE COMPANY
    </strong>
    : In no event shall the Company be liable of any direct, indirect, incidental, special, exemplary, or consequential damages (including, but not limited to, procurement of substitute goods or services; loss of use, data, or profits; or business interruption).
  </p>
  <p>
    <strong>
      NO INVOLVEMENT OF COMPANY IN ANY TRANSACTIONS BETWEEN USERS
    </strong>
    : WHILE THE COMPANY DEVELOPED THE MELON PROTOCOL, IT IS IMPORTANT TO NOTE THAT THE COMPANY IS NOT INVOLVED IN ANY OF THE INTERACTIONS BY MANAGERS WITH THE MELON PROTOCOL, NOR DOES IT HOLD CUSTODY OF ANY OF THE USER ASSETS INVOLVED DURING THE COMPETITION.
  </p>
  <p>
    <strong>
      RESPONSIBILTY FOR SECURITY OF PASSWORDS REMAINS SOLELY WITH USER
    </strong>
    : There is no possibility for the Company to recover USER’S private password, respectively USER’S private key. When setting up an account within the Melon platform, user will be responsible for keeping user’s own account secrets, which may be a twelve-word seed phrase, an account file, or other locally stored secret information. The Melon Protocol encrypts this information locally with a password user provides, that is never sent to the Company servers, nor known otherwise to the Company. The Company cannot and will not be liable for any loss or damage arising from users failure to comply with this section.
  </p>
  <p>
    <strong>
      INTERACTIONS WITH PROTOCOL CANNOT BE REVISED 
    </strong>
    : User understands there are no rights of recourse or compensation whatsoever especially with regards to any complaints of theft, malfunctioning of the Melon Protocol, external abuse of the Melon Protocol, misbehavior of managers etc.
  </p>
  <p>
    <strong>
      RESPONSIBILITY FOR LEGAL COMPLIANCE AND TAXATION SOLELY WITH USER
    </strong>
    : User understands and accepts that user is solely responsible for compliance with all laws that may apply to user’s particular use of the Melon Protocol. Company makes no representation regarding the application of any laws. User is aware of the fact, that the use of the Melon Protocol may be qualified differently in different jurisdictions and agrees that Company is not responsible for determining whether or which laws may apply to his/her use of the Melon Protocol. Further, user understands and accepts that it is user’s sole responsibility to assess whether any taxes will become due for his/her interactions with the Melon Protocol under any applicable jurisdiction.
  </p>
  <p>
    <strong>
      JURISDICTION AND GOVERNING LAW
    </strong>
    : Any disputes arising out of this Disclaimer and the use of the Melon Protocol shall be submitted to the exclusive jurisdiction of the courts of Zug, Switzerland and this Disclaimer, as well as the use of the Melon Protocol shall be exclusively governed by Swiss law, except for the United Nations Convention for the International Sales of Goods which is excluded.
  </p>
  <p>
    BY SIGNING THE HASH OF
     
    <strong>
      <a
        href="https://github.com/melonproject/contribution/blob/master/Melonport_Second%20Contribution%20Period_FINAL.pdf"
        rel="noopener noreferrer"
        target="_blank"
      >
        THIS DOCUMENT
      </a>
       
    </strong>
     
    THE USER UNDERSTANDS AND ACCEPTS ALL THE TERMS AND CONDITIONS AS SET HEREIN.
  </p>
  <Button
    onClick={[Function]}
    style="secondary"
  >
    I understand and agree to sign the above terms and conditions.
  </Button>
</div>
`;

exports[`TermsAndConditions should render correctly without networkId 1`] = `
<div
  className="conditions"
>
  <h4>
    TERMS AND CONDITIONS
  </h4>
  <p>
    The following terms and conditions have been deployed to IPFS and resolve to the following hash:
     
    <a
      href="https://ipfs.io/ipfs/QmQrY13NWoSwK2Ani5eBwiyo1gsNa4xLQYfS8zBuHro8Lm"
      rel="noopener noreferrer"
      target="_blank"
    >
      QmQrY13NWoSwK2Ani5eBwiyo1gsNa4xLQYfS8zBuHro8Lm
    </a>
  </p>
  <p>
    <strong>
      THIS SOFTWARE IS INTENDED FOR TESTING PURPOSES ONLY.
    </strong>
  </p>
  <p>
    It is intended to be used on the Kovan testnet and is by no means intended to be used on the main net. Never send assets or tokens of real value to a fund contract on the test net; they could be impossible to recover.
  </p>
  <p>
    <strong>
      DO NOT USE REAL ETHER OR VALUE TOKEN.
    </strong>
  </p>
  <p>
    The User understands and accepts that in the event of announcing any Melon related competitions, the winners and prizes will be determined at full discretion by Melonport AG and Melonport AG alone.
  </p>
  <p>
    The User understands and accepts that this software is provided “as is” and any expressed or implied warranties of merchantability and fitness for a particular purpose are disclaimed. In no event shall Melonport AG or its Contributors be liable of any direct, indirect, incidental, special, exemplary, or consequential damages (including, but not limited to, procurement of substitute goods or services; loss of use, data, or profits; or business interruption).
  </p>
  <p>
    Neither Melonport AG nor its Contributors are liable for any damages, however caused and through any theory of liability, whether in contract, strict liability, or tort (including negligence or otherwise) arising in any way out of the use of this software, even if advised of the possibility of such damage.
  </p>
  <p>
    This Agreement, shall be governed by, interpreted and construed in accordance with the substantive laws of Switzerland. The competent courts of Zug, Switzerland, shall have exclusive jurisdiction for all disputes arising out of or in connection with this Agreement, subject to appeal, if any.
  </p>
  <p>
    <strong>
      BY SIGNING THE HASH OF THIS DOCUMENT THE USER UNDERSTANDS AND ACCEPTS ALL THE TERMS AND CONDITIONS AS SET HEREIN.
    </strong>
  </p>
  <Button
    onClick={[Function]}
    style="secondary"
  >
    I understand and agree to sign the above terms and conditions.
  </Button>
</div>
`;
